### DISCUSSION TITLE
How to perform this very simple object movement?

### DISCUSSION THREAD

**Discussion by AlexandrosGounis** [pmndrs/use-cannon#214] [Q&A] [answered]
See demo:
https://tp1r8.sse.codesandbox.io/

A raycaster (using onpointermove event) moves the stick to different positions.

Using the use-cannon api, that's
```ts
const [pawnRef, pawnApi] = useCylinder(...)
...
function onMove(e) {
    pawnApi.position.set(...)
}
```

As you can see, we are not supposed to move `kinematic` objects like that. Most collisions do not register unless you are very slow with the stick movements. Changing the velocity does not work either, it looks like the sticks jumps around or stutters. 
Is there any way to do this with cannon and use-cannon?

**Comment by drcmda** [comment, marked as answer]
from what i have read cannon misses something, i don't know the technical details, but it can't detect collisions if the object moves too fast. i had the sam trouble with some mini games i've made, ping pong and arkanoid, it doesn't really lend itself to that. but i also don't know how much it would cost to implement it in cannon. @stockHuman @marcofugaro

**Reply by AlexandrosGounis** [reply]
I was actually able to resolve it by increasing the `step` to `1/240` and by directly mutating the velocity of the object that I wanted to move, instead of its position. It feels harder than it should, though, and the same movement is quite easy to implement in Swift. It is probably cannon making our life difficult, but I know that ammo, or oimo have other issues too.
Thanks for your support @drcmda
